client/my-sites/stats/controller.js

```diff
@@ -188,7 +189,12 @@ module.exports = {
 
 		analytics.pageView.record( basePath, analyticsPageTitle + ' > Insights' );
 
-		if ( site.post_count <= 1 &&
+		// Calculate the number of posts created by the user (not Headstarted) to determine if we should show the Insights tab, or the NUX placeholder.
+		postCount = site.post_count;
+		if ( site.options.headstart ) {
+			postCount = site.post_count - site.options.headstart.original.post.length;
```

Comment:
> This also assumes that site.post_count always exists and is an integer

Yep, this is true (as long as the user is logged in, which they need to be to be viewing Calypso).